main.go

```diff
@@ -111,6 +111,7 @@ func processCommand() int {
 			// override default ssh-agent socket
 			os.Setenv("SSH_AUTH_SOCK", agent.SocketFile())
 			log.Debugf("set SSH_AUTH_SOCK to %q\n", agent.SocketFile())
+			sshClient.PrependArgs([]string{"-o", fmt.Sprintf("IdentityAgent=%s", agent.SocketFile())})
```

Comment:
Yes I had the same though but then I remembered that `-o ProxyCommand=...` exists and something might rely on it. I did not find any disadvantages to continue to set `SSH_AUTH_SOCK` so I kept it just in case.

I'm fine either way to be honest.

Reply:
Let's go with the minimalist DRY approach, and separately I'll look at adding some magic to avoid prepending the additional argument when it's not required:

```suggestion
			sshClient.PrependArgs([]string{"-o", "IdentityAgent=SSH_AUTH_SOCK"})
```